python/pyarrow/_dataset.pyx

```diff
@@ -953,7 +953,7 @@ cdef class InMemoryDataset(Dataset):
     ----------
     source : RecordBatch, Table, list, tuple
         The data for this dataset. Can be a RecordBatch, Table, list of
-        RecordBatch/Table, iterable of RecordBatch, or a RecordBatchReader
+        RecordBatch/Table, or an iterable of RecordBatch
```

Comment:
Hmm isn't a `RecordBatchReader` an iterable of `RecordBatch`? If a RBR doesn't work, that would suggest an iterable wouldn't either.

Reply:
That is also my read of a `RecordBatchReader`, but this code path suggest that a list or tuple of batches would work:

```
        if isinstance(source, (list, tuple)):
            batches = []
            for item in source:
                if isinstance(item, pa.RecordBatch):
batches.append(item)
```